src/ol/color.js

```diff
@@ -23,6 +20,81 @@ import {clamp} from './math.js';
  */
 export const NO_COLOR = [NaN, NaN, NaN, 0];
 
+let colorParseContext;
+/**
+ * @return {CanvasRenderingContext2D} The color parse context
+ */
+function getColorParseContext() {
+  if (!colorParseContext) {
+    colorParseContext = createCanvasContext2D(1, 1, undefined, {
+      willReadFrequently: true,
+      desynchronized: true,
+    });
+  }
+  return colorParseContext;
+}
+
+const rgbRegEx =
+  /^rgba?\(\s*([0-9]+%?)[\s,]+([0-9]+%?)[\s,]+([0-9]+%?)[\s/,]*([0-9.%]+)?\s*\)$/i;
```

Comment:
This should reject all invalid cases (but does still not). If something valid is rejected here it will be correctly handled at the end of the parseRgba function.

```suggestion
  /^rgba?\((\d+%?)(?:\s|,\s?)(\d+%?)(?:\s|,\s?)(\d+%?)(?:\s?[\s\/,]\s?((?:\d+\.\d+|\.?\d+)%?))?\)$/i;
```
Cases that are still handled incorrectly (will parse a color from invalid input):
`rgb(0%,0,0)` % not allowed with commas
`rgb(0 0,0)` mix of comma and space as separators
`rgba(0,0,0/0)` mix of comma and slash as separators

Reply:
I split this up into 3 regular expressions, so now also the cases you mentioned are handled correctly. I added those to the tests as well.